### ISSUE TITLE
How to set http port & force client to use http not grpc

### ISSUE THREAD

**Issue by samcov** [dapr/dotnet-sdk#609]
await client.SaveStateAsync(storeName, stateKeyName, state, cancellationToken: cancellationToken);

I'm getting this crashing error, but I don't want to use grpc, **I want to use http**.

  | Message | "Status(StatusCode=\"Internal\", Detail=\"Error starting gRPC call. HttpRequestException: No connection could be made because the target machine actively refused it. SocketException: No connection could be made because the target machine actively refused it.\", DebugException=\"System.Net.Http.HttpRequestException: No connection could be made because the target machine actively refused it.\r\n ---> System.Net.Sockets.SocketException (10061): No connection could be made because the target machine actively refused it.\r\n   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)\r\n   --- End of inner exception stack trace ---\r\n   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)\r\n   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)\r\n   at System.Net.Http.DiagnosticsHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)\r\n   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)\")" | string

**Comment by rynowak** [comment]
We don't provide an option to configure what protocol the SDK does (except service invocation). Can you tell us more about the scenario and why you think this is important?

**Comment by samcov** [comment]
**Yes, but first an update**: I wasn't using the standard grpc port, but once I did, I get a different error.
**New Error**: Status(StatusCode="Unimplemented", Detail="unknown service dapr.proto.runtime.v1.Dapr")

This may have something to do with something I'm missing in the method definitions somewhere, not sure.

**Why I think this is important**: After understanding that the sdk is GRPC only, it doesn't really matter to me, although for some context, the eShopOnDapr uses the sdk for most things, but goes to the http api for flexibility they say. Having said that, for my purposes, GRPC is great, I'll take the performance... thank you :)

**My Setup**: It's a bit different than what most use, but I like not having to install dapr locally. I run dapr in a docker container that exposes the ports to my local machine which allows me to be a lot more portable and share with others.

Everything appears to work, especially the REST Api, so now I'd like to get the GRPC working(I can show the docker-compose if needed)

**Comment by rynowak** [comment]
If you want help troubleshooting, please share code/examples. At this point I need more info about what you're trying to accomplish :)

**Comment by samcov** [comment]
**Here's the code I shared**: 
**I ran this**:  await client.SaveStateAsync(storeName, stateKeyName, state, cancellationToken: cancellationToken);

**I got this**:  Status(StatusCode="Unimplemented", Detail="unknown service dapr.proto.runtime.v1.Dapr")

What does this error mean is the question since it only happens when using the SDK, it works with the REST call.

**Comment by rynowak** [comment]
> Status(StatusCode="Unimplemented", Detail="unknown service dapr.proto.runtime.v1.Dapr")

If I had to guess I think it means that the SDK is trying to talk to something that isn't the `daprd` process. This is the first time someone has reported a problem like this, so I'm not sure what it is.

Can you give more info about your setup? You mentioned that you are running Dapr in a container, and then you configured the SDK to talk to it.  Can you share an example of how your are running Dapr (docker CLI example, or docker-compose `.yaml`)? Can you share how you are configuring the SDK to locate it? I suspect that's where the issue is.

**Comment by samcov** [comment]
It's been reported before, but no clear reason or solution was offered: https://github.com/dapr/dapr/issues/1635

**Docker-compose**:
version: '3'
services:
  **placement**:
    image: daprio/dapr:1.0.0
    command: [ "./placement","-port","50001" ]
    ports:
      - "50001:50001"
  **redis**:
    image: "redis:alpine"
    ports:
      - "6379:6379"
    expose:
      - "6379"
    
  **sam-dapr**:
    image: "daprio/daprd:1.0.0"
    command: ["./daprd",
    "-app-id", "myapp",
    "-dapr-http-port", "3500",
    "-components-path", "/components"]
    volumes:
        - "./components/:/components"
    ports:
      - "3500:3500"
    expose:
      - "3500"
    depends_on:
      - redis

**Comment by rynowak** [comment]
So for this approach to work, you'll need to expose a port for grpc to and from the sidecar, and then configure that in your application or via the `DAPR_GRPC_PORT`

**Comment by samcov** [comment]
I'll test this and get back to you.

**Comment by samcov** [comment]
**It Worked!!!** You're the man!

It kept dying at first, but then I noticed I was exposing the same port, 50001 to the host for two different containers, so I altered the placement service, which I don't need to talk to now as shown below, everything remained the same.

Essentially you were correct, I only allowed GRPC communication one way, opening that port to the app allowed 2 way communication.

**placement**:
image: daprio/dapr:1.0.0
command: [ "./placement","-port","50001" ]
ports:
- "**50002**:50001"
